docs/api/types.md

```diff
@@ -171,14 +171,67 @@ type Notification = {
     | "mod_add";
 
   // The content of the notification. The structure of this object will vary based on the type of notification.
-  notif: {};
+  notif:
+    | newVotesNotif
+    | deletedPostNotif
+    | newCommentNotif
+    | commentReplyNotif
+    | newBadgeNotif
+    | modAddNotif;
 
   seen: boolean; // Whether the notification was seen by the authenticated user.
   seenAt: time | null; // If the notification was seen, the time at which it was seen, otherwise null.
   createdAt: time; // The time at which the notification was created.
 };
+
+type newVotesNotif = {
+	noVotes: number; // The number of votes the post or comment received.
+	targetId: string; // The ID of the post or comment that was voted on.
+	targetType: "post" | "comment"; // The type of the target.
+	post: Post | Comment; // The target item, either of type `Comment` if `targetType` is "comment" or `Post` otherwise.
+	comment: Comment; // Only present if the `targetType` is "comment".
+}
+
+type deletedPostNotif = {
+	deletedAs: "mods" | "admins"; // The user group that deleted the post.
+	post: Post; // The post that was deleted.
+	targetId: string; // The ID of the post that was deleted.
+	targetType: "post" | "comment"; // The type of the target.
+}
+
+type newCommentNotif = {
+	commentAuthor: string; // The username of the comment author.
+	commentId: string; // The ID of the comment.
+	firstCreatedAt: time; // The time at which the first comment was created.
+	noComments: number; // The number of new comments on the post.
+	post: Post; // The parent post.
+	postId: string; // The ID of the post.
+}
+
+type commentReplyNotif = {
+    commentAuthor: string; // The username of the comment author.
+    commentId: string; // The ID of the comment.
+    firstCreatedAt: time; // The time at which the first comment was created.
+    noComments: number; // The number of comments on the post.
```

Comment:
Be consistent between tabs and spaces for indentation. As the rest of the docs follow a 2 space indentation, I'd say follow that.